libsolidity/ast/AST.h

```diff
@@ -1150,6 +1151,23 @@ class TupleExpression: public Expression
 	std::vector<ASTPointer<Expression>> m_components;
 };
 
+class InlineArrayExpression: public Expression
```

Comment:
This has a lot of code in common with `TupleExpression`. Can we just add a `bool` flag to the constructor of `TupleExpression` telling whether it is a tuple or array?